dev/requirements.txt

```diff
@@ -4,3 +4,5 @@ PyGithub==1.26.0
 Unidecode==0.04.19
 sphinx
 pydata_sphinx_theme
+ipython
```

Comment:
Are there other implications to this, like, our python packaging now requires ipython when installing pyspark? (sorry ignorant question)

Reply:
Nope, it means nothing special and this file is used nowhere within Spark. It's just for some dev people but I doubt if this is actually often used. We should leverage this file and standardize the dependencies somehow like @nchammas tried before but .. I currently have a good idea about how to handle it.